root/account/applications/Index.js

```diff
@@ -0,0 +1,134 @@
+import commaOnlyList from '../../static/scripts/common/i18n/commaOnlyList';
+import loopParity from '../../utility/loopParity';
+
+type Props = {|
+  +applications: $ReadOnlyArray<ApplicationT>,
+  +appsPager: PagerT,
+  +tokens: $ReadOnlyArray<EditorOAuthTokenT>,
+  +tokensPager: PagerT,
+|};
+
+const buildApplicationRow = (application: ApplicationT, index: number) => (
+  <tr className={loopParity(index)} key={application.id}>
+    <td>{application.name}</td>
+    <td>
+      {application.oauth_redirect_uri ? l('Web Application') : l('Installed Application')}
+    </td>
+    <td><code>{application.oauth_id}</code></td>
+    <td><code>{application.oauth_secret}</code></td>
+    <td>
+      <a href={'/account/applications/edit/' + application.id}>{l('Edit')}</a>
+      {' | '}
+      <a href={'/account/applications/remove/' + application.id}>{l('Remove')}</a>
+    </td>
+  </tr>
+);
+
+const buildTokenRow = (token: EditorOAuthTokenT, index: number) => (
+  <tr className={loopParity(index)} key={token.id}>
+    <td>{token.application.name}</td>
+    <td>{formatScopes(token)}</td>
+    <td>
+      <a href={'/account/applications/revoke-access/' + token.application.id + '/' + token.scope}>
+        {l('Revoke Access')}
+      </a>
+    </td>
+  </tr>
+);
+
+function formatScopes(token: EditorOAuthTokenT) {
+  const lScopes = [];
+
+  token.permissions.forEach((perm) => {
+    lScopes.push(l(perm));
+  });
+
+  if (token.is_offline) {
+    lScopes.push(l('Offline Access'));
+  }
+
+  lScopes.sort(compare);
+
+  return commaOnlyList(lScopes);
+}
+
+const Index = ({applications, appsPager, tokens, tokensPager}: Props) => (
+  <Layout fullWidth title={l('Applications')}>
+    <h1>{l('Applications')}</h1>
+
+    <h2>{l('Authorized Applications')}</h2>
+
+    <p>
+      {l('Some applications and websites support accessing private data from or submitting data to MusicBrainz but require your permission to access your account. These are the applications that you have authorized to access your MusicBrainz account. If you no longer use some of the applications, you can revoke their access.')}
+    </p>
+
+    {tokens.length
+      ? (
+        <PaginatedResults pageVar="tokens_page" pager={tokensPager}>
+          <table className="tbl">
+            <thead>
+              <tr>
+                <th>{l('Application')}</th>
+                <th>{l('Access')}</th>
+                <th>{l('Actions')}</th>
+              </tr>
+            </thead>
+            <tbody>
+              {tokens.map(buildTokenRow)}
+            </tbody>
+          </table>
+        </PaginatedResults>
+      ) : (
+        <p>{l('You have not authorized any applications.')}</p>
+      )}
+
+    <h2>{l('Developer Applications')}</h2>
+
+    <p>
+      {l('Do you want to develop an application that uses the {ws|MusicBrainz web service}? {register|Register an application} to generate OAuth tokens. See our {oauth2|OAuth documentation} for more details.', {
+        __react: 'true',
```

Comment:
Small nit, it should be a boolean without quotes (here and elsewhere). It doesn't really matter except that it might not type-check later, and I think uglify can minify `true` to `!1`.

Reply:
Fixed at https://github.com/metabrainz/musicbrainz-server/pull/699/commits/79ae598a8cec954a67da07209377d94c74ae508a#diff-4c3d028cef4917d20675baed1fb0ce28R99 and everywhere else